flink-sql-connector-opensearch/pom.xml

```diff
@@ -0,0 +1,159 @@
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-connector-opensearch-parent</artifactId>
+		<version>1.0.0-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-sql-connector-opensearch</artifactId>
+	<name>Flink : Connectors : SQL : Opensearch</name>
+
+	<packaging>jar</packaging>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-opensearch</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<artifactSet>
+								<includes>
+									<include>*:*</include>
+								</includes>
+								<excludes>
+									<!-- These dependencies are not required. -->
+									<exclude>com.tdunning:t-digest</exclude>
+									<exclude>joda-time:joda-time</exclude>
+									<exclude>net.sf.jopt-simple:jopt-simple</exclude>
+									<exclude>net.java.dev.jna:jna</exclude>
+									<exclude>org.hdrhistogram:HdrHistogram</exclude>
+									<exclude>org.yaml:snakeyaml</exclude>
+								</excludes>
+							</artifactSet>
+							<filters>
+								<!-- Unless otherwise noticed these filters only serve to reduce the size of the resulting
+									jar by removing unnecessary files -->
+								<filter>
+									<artifact>org.opensearch:opensearch</artifact>
+									<excludes>
+										<exclude>config/**</exclude>
+										<exclude>modules.txt</exclude>
+										<exclude>plugins.txt</exclude>
+										<exclude>org/joda/**</exclude>
+										<!-- Exclude bootstrap due to license issues -->
+										<exclude>org/opensearch/bootstrap/**</exclude>
+									</excludes>
+								</filter>
+								<filter>
+									<artifact>org.opensearch.client:opensearch-rest-high-level-client</artifact>
+									<excludes>
+										<exclude>forbidden/**</exclude>
+									</excludes>
+								</filter>
+								<filter>
+									<artifact>org.apache.httpcomponents:httpclient</artifact>
+									<excludes>
+										<exclude>mozilla/**</exclude>
+									</excludes>
+								</filter>
+								<filter>
+									<artifact>org.apache.lucene:lucene-analyzers-common</artifact>
+									<excludes>
+										<exclude>org/tartarus/**</exclude>
+									</excludes>
+								</filter>
+								<filter>
+									<artifact>*:*</artifact>
+									<excludes>
+										<!-- exclude Java 9 specific classes as otherwise the shade-plugin crashes -->
+										<exclude>META-INF/versions/**</exclude>
```

Comment:
This seems odd; Flink itself is bundling java version-specific files.

Can you show me the error message you got?

Reply:
This one is taken from https://github.com/apache/flink-connector-elasticsearch/blob/main/flink-sql-connector-elasticsearch7/pom.xml#L117 since OpenSearch is a fork of Elasticseach 7.10.2

Reply:
What happens without it? :see_no_evil:

Reply:
Could be quite outdated, removed (for now)

Reply:
> What happens without it? see_no_evil

@zentol answer to your (and mine) curiosity:

```
 Error:  Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.1:shade (shade-flink) on project flink-sql-connector-opensearch: Error creating shaded jar: Problem shading JAR /home/runner/.m2/repository/org/opensearch/opensearch/1.3.0/opensearch-1.3.0.jar entry META-INF/versions/11/org/opensearch/monitor/jvm/JvmPid.class: java.lang.IllegalArgumentException -> [Help 1]
``

Reply:
Alright then. :sweat_smile: